software/controller/lib/hal/adc.cpp

```diff
@@ -261,31 +257,44 @@ static constexpr int AdcConversionTime = [] {
   __builtin_unreachable();
 }();
 
-/// \TODO: have caller provide mappings in a different layer
+// local helper functions to manipulate ADC registers
```

Comment:
These abstractions are great, but maybe we ought to have the signatures in the header, if even as private static functions?